### PR TITLE
PNDA-4537: Metrics retention needs to be set per-flavor

### PR DESCRIPTION
**Pull request by Hitesh-Maplelabs**
**Problem Statement:**
PNDA-4537: Metrics retention needs to be set per-flavor

**Changes:**
1. Set retention settings in Pillar for each flavor.
2. Updated default list of metrics to capture for spark applications.

**Test details:**
AWS-RHEL-PICO-HDP

### CODE REVIEW

File: pillar/flavors/pico.sls [hunk 1]
```diff
@@ -45,6 +45,9 @@
             "parallelism": 1,
             "taskmanager_mem_preallocate": false,
             "pyflink_yarn_container_count": 1
+        },
+        "graphite-api":{
+            "retentions": "60:10080"
```
Comment (trsmith2) [inline pillar/flavors/pico.sls]: Can we use the singular form here 'retention' and note what it is a retention for in the name (i.e. apps)

Pico should have the lowest retention of the three flavors. A day is sufficient.

File: pillar/flavors/production.sls [hunk 2]
```diff
@@ -45,6 +45,9 @@
             "parallelism": 1,
             "taskmanager_mem_preallocate": false,
             "pyflink_yarn_container_count": 1
+        },
+        "graphite-api":{
+            "retentions": "60:10080"
```
Comment (trsmith2) [inline pillar/flavors/production.sls]: Production should have the highest retention of the three flavors.

File: pillar/flavors/standard.sls [hunk 3]
```diff
@@ -45,6 +45,9 @@
             "parallelism": 1,
             "taskmanager_mem_preallocate": false,
             "pyflink_yarn_container_count": 1
+        },
+        "graphite-api":{
+            "retentions": "60:10080"
```
Comment (trsmith2) [inline pillar/flavors/standard.sls]: Standard should have a retention between that of production and pico.

**Comment by trsmith2** [comment]
The list of metrics is much better, but can you please address the minor comments on the retention settings.

File: pillar/flavors/pico.sls (hunk 1 above)
Comment (Hitesh-Maplelabs) [reply pillar/flavors/pico.sls]: Changed 'retentions' to 'retention_spark_metrics'.
For pico, have set retention a day.

File: pillar/flavors/production.sls (hunk 2 above)
Comment (Hitesh-Maplelabs) [reply pillar/flavors/production.sls]: For production, have set retention 14 days.

File: pillar/flavors/standard.sls (hunk 3 above)
Comment (Hitesh-Maplelabs) [reply pillar/flavors/standard.sls]: For standard, have set retention 7 days.

**Comment by trsmith2** [comment]
Thanks for the updates!

**Pull request merged**